pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -350,6 +355,61 @@ func (a *AmbientIndex) insertWorkloadToService(svcAddress string, workload *mode
 	a.byService[svcAddress] = append(a.byService[svcAddress], workload)
 }
 
+func (a *AmbientIndex) updateWaypoint(sa types.NamespacedName, ipStr string, isDelete bool) map[model.ConfigKey]struct{} {
+	addr := netip.MustParseAddr(ipStr).AsSlice()
```

Comment:
nit: we should probably avoid MustParse. I know this is coming from k8s, but effectively the same as .unwrap.